tt_metal/impl/dispatch/command_queue.hpp

```diff
@@ -351,24 +341,70 @@ class CommandQueue {
 
     void launch(launch_msg_t& msg);
 
-    friend void EnqueueReadBuffer(CommandQueue& cq, std::variant<std::reference_wrapper<Buffer>, std::shared_ptr<Buffer>> buffer, vector<uint32_t>& dst, bool blocking);
-    friend void EnqueueWriteBuffer(CommandQueue& cq, std::variant<std::reference_wrapper<Buffer>, std::shared_ptr<Buffer>> buffer, vector<uint32_t>& src, bool blocking);
-    friend void EnqueueReadBuffer(CommandQueue& cq, std::variant<std::reference_wrapper<Buffer>, std::shared_ptr<Buffer>> buffer, void* dst, bool blocking);
-    friend void EnqueueWriteBuffer(CommandQueue& cq, std::variant<std::reference_wrapper<Buffer>, std::shared_ptr<Buffer>> buffer, const void* src, bool blocking);
-    friend void EnqueueProgram(CommandQueue& cq, Program& program, bool blocking, std::optional<std::reference_wrapper<Trace>> trace);
-    friend void Finish(CommandQueue& cq);
-    friend void detail::EnqueueRestart(CommandQueue& cq);
-    friend void ClearProgramCache(CommandQueue& cq);
-    friend CommandQueue &detail::GetCommandQueue(Device *device);
-
-    // Trace APIs
-    friend Trace BeginTrace(CommandQueue& command_queue);
-    friend void EndTrace(Trace& trace);
-    friend void EnqueueTrace(Trace& trace, bool blocking);
-    friend class Device;
-    friend class Trace;
+    SystemMemoryManager& sysmem_manager() { return this->sysmem_manager_; }
```

Comment:
I don't use this pattern. No suffix and no helper function.

Reply:
so you want me to make the member public? I cannot compile if I have a function and member both named the same thing

Reply:
or just make CQ a friend of device.